subsys/net/lib/sockets/getaddrinfo.c

```diff
@@ -105,7 +105,7 @@ static int exec_query(const char *host, int family,
 {
 	enum dns_query_type qtype = DNS_QUERY_TYPE_A;
 
-	if (IS_ENABLED(CONFIG_NET_IPV6) && family != AF_INET) {
+	if (IS_ENABLED(CONFIG_NET_IPV6) && family == AF_INET6) {
```

Comment:
Probably worth adding a comment, that for AF_UNSPEC case, we query IPv4 in this func.